### PR TITLE
Add improvements.

### PR DESCRIPTION
**Pull request by dewniMW**

### CODE REVIEW

File: en/docs/references/concepts/authentication/dcr.md [hunk 1]
```diff
@@ -2,13 +2,12 @@
...
  
-
-## How it works 
+## How does it work? 
 
 To register a new client at the authorization server:
 
 1. Client discovers the client registration endpoint. The registration endpoint URL is discoverable via WebFinger.
-   For more information on discovering the client registration endpoint, see [OpenID Connect Discovery](../login/discovery.md).
+   For more information on discovering the client registration endpoint, see [OpenID Connect Discovery](discovery.md).
```
Comment (gomathyk) [inline en/docs/references/concepts/authentication/dcr.md:10]: Did we have to mention the .md part to make it work?

File: en/docs/references/concepts/authentication/dcr.md [hunk 2]
```diff
@@ -17,23 +16,28 @@ To register a new client at the authorization server:
...
 - A client can be registered dynamically with the authorization server itself
 - A programmer can register a client programmatically
 
 
-### What is DCRM (Dynamic Client Registration Management) ?
+### What is DCRM (Dynamic Client Registration Management)?
 
-DCRM is an extension to the DCR, which introduced form this [specification](https://tools.ietf.org/html/rfc7592). 
-Main functionalities which specified are :
+DCRM is an extension to the DCR, introduced from this [specification](https://tools.ietf.org/html/rfc7592).
```
Comment (gomathyk) [inline en/docs/references/concepts/authentication/dcr.md:35]:
```suggestion
DCRM is an extension to the DCR, introduced from [this specification](https://tools.ietf.org/html/rfc7592). 
```

File: en/docs/references/concepts/authentication/discovery.md [hunk 3]
```diff
@@ -38,23 +39,23 @@ of the OpenID Provider. Following information is required when making a request
...
- ``` https://[server-url]/.well-known/webfinger```
+ ``` https://<SERVER_URL>/.well-known/webfinger```
  
-##  Expose Metadata
+### Expose metadata
  
  The OpenID provider exposes a JSON document listing all the metadata of the server. 
- This document contains of standard endpoints and other server supported meta data values. 
- The target audience of this file is the client applications developers who send requests to the
+ This document contains of standard endpoints and other server supported metadata values.
```
Comment (gomathyk) [inline en/docs/references/concepts/authentication/discovery.md:50]:
```suggestion
 This document contains standard endpoints and other server supported metadata values. 
```

File: en/docs/references/concepts/authentication/discovery.md [hunk 4]
```diff
@@ -38,23 +39,23 @@ of the OpenID Provider. Following information is required when making a request
...
+ ``` https://<SERVER_URL>/.well-known/webfinger```
  
-##  Expose Metadata
+### Expose metadata
  
  The OpenID provider exposes a JSON document listing all the metadata of the server. 
- This document contains of standard endpoints and other server supported meta data values. 
- The target audience of this file is the client applications developers who send requests to the
+ This document contains of standard endpoints and other server supported metadata values. 
+ The target audience of this file is the client application developers who send requests to the
```
Comment (gomathyk) [inline en/docs/references/concepts/authentication/discovery.md:51]:
```suggestion
 The target audience of this file are the client application developers who send requests to the
```

File: en/docs/references/concepts/authentication/discovery.md [hunk 5]
```diff
@@ -38,23 +39,23 @@ of the OpenID Provider. Following information is required when making a request
...
  server.
  
- Following endpoint responsibles for exposing the server's metadata.
+ The following endpoint is responsible for exposing the server's metadata.
  
- ```https://[server-url]/.well-known/openid-configuration```
+ ```https://<SERVER_URL>/.well-known/openid-configuration```
  
- A sample JSON file that will contain server meta data is shown below.
+ A sample JSON file that will contain server metadata is shown below.
```
Comment (gomathyk) [inline en/docs/references/concepts/authentication/discovery.md:58]:
```suggestion
 A sample JSON file that containing the server metadata is shown below.
```

File: en/docs/references/concepts/authentication/jwks.md [hunk 6]
```diff
@@ -1,72 +1,80 @@
...
-### How it works ?
-Following sequence diagram illustrates the scenario where a JWT obtained
-from a third party IDP is validated using the JWKS Based JWT Validator.
+---
+
+## How does it work?
 
-![jwks-validation-flow]( ../../assets/img/concepts/jwks-validation-flow.png) 
+The following sequence diagram illustrates the scenario where a JWT obtained
+from a third party IdP is validated using the JWKS based JWT Validator.
```
Comment (gomathyk) [inline en/docs/references/concepts/authentication/jwks.md:24]:
```suggestion
from a third party IdP is validated using the JWKS-based JWT Validator.
```

File: en/docs/references/concepts/authentication/session-management.md [hunk 7]
```diff
@@ -1,92 +1,94 @@
 # OpenID Connect Session Management
 
-[The OpenID Connect Session Management specification](https://openid.net/specs/openid-connect-session-1_0.html)
-defines methodologies to manage user sessions and log-out end-users at the authorization server using front-channel communication.
-In this approach the login/logout requests from the RP (Relying Party or the client application) to OP (OpenID Provider)
-and OP to RP is done via the User Agent (browser based).
+The OpenID Connect Session Management [specification](https://openid.net/specs/openid-connect-session-1_0.html)
+defines methodologies to manage user sessions and logout end-users at the authorization server using front-channel communication.
```
Comment (gomathyk) [inline en/docs/references/concepts/authentication/session-management.md:4]:
```suggestion
defines methodologies to manage user sessions and log out end-users at the authorization server using front-channel communication.
```

File: en/docs/references/concepts/authentication/session-management.md [hunk 8]
```diff
@@ -1,92 +1,94 @@
...
+The following diagram represents the flow.
 
 ![session-management](../../../assets/img/concepts/session-management.png)
 
-- Once the end-user needs to login to the RP, the RP sends an authentication request to the OP.
+- Once the end-user needs to log in to the RP, the RP sends an authentication request to the OP.
 - The OP responds back with the **session_state**.
 - The RP iframe continuously polls the OP iframe to detect any state changes.
-- The OP iframe responds back with one of the following status.
+- The OP iframe responds back with one of the following statuses.
```
Comment (gomathyk) [inline en/docs/references/concepts/authentication/session-management.md:53]:
```suggestion
- The OP iframe responds with one of the following statuses.
```

File: en/docs/references/concepts/authentication/session-management.md [hunk 9]
```diff
@@ -1,92 +1,94 @@
...
-  so as to not cause potential infinite loops that generate network traffic to the OP, so it directly logged the user out.
+- changed: This indicates that user-session is still valid at the OP. RP will continue to poll
+             OP iframe to detect any session changes. 
+- unchanged: This indicates that the session has changed at the OP.
+               This can happen due to user logout, session timeout, or a user logging in from a different client application.
+               Upon receipt of the `changed` state, the RP performs re-authentication with prompt=none 
+               to obtain the current session state at the OP.
+- error: This indicates the message sent was determined by the OP to be malformed.
+           Upon receipt of error, the RP must not perform re-authentication with `prompt=none`,
+           so as to not cause potential infinite loops that generate network traffic to the OP, so it directly logged the user out.
```
Comment (gomathyk) [inline en/docs/references/concepts/authentication/session-management.md:65]:
```suggestion
           so as to not cause potential infinite loops that generate network traffic to the OP, it directly logs the user out.
```

File: en/docs/references/concepts/authentication/traditional-authentication-request.md [hunk 10]
```diff
@@ -31,7 +31,7 @@ The parameter passed from the application to the identity provider to maintain a
 
 | Parameter vale                 | Description    | 
 | --------------------- | ------------- |
-| prompt= none |The Authorization Server does not display any authentication or consent user interface pages. An error is returned if an end user is not already authenticated or the client does not have pre-configured consent for the requested claims or if there are any other unfulfilled conditions for processing the request.  |                       
+| prompt= none |The authorization server does not display any authentication or consent user interface pages. An error is returned if an end user is not already authenticated or the client does not have pre-configured consent for the requested claims or if there are any other unfulfilled conditions for processing the request.  |
```
Comment (gomathyk) [inline en/docs/references/concepts/authentication/traditional-authentication-request.md:34]:
```suggestion
| prompt= none |The authorization server does not display any authentication or consent user interface pages. An error is returned if an end user is not already authenticated, the client does not have pre-configured consent for the requested claims, or if there are any other unfulfilled conditions for processing the request.  |                       
```

**Review by gomathyk — CHANGES_REQUESTED** [review]

File: en/docs/references/concepts/authentication/dcr.md (hunk 1 above)
Comment (dewniMW) [reply en/docs/references/concepts/authentication/dcr.md:10]: yes it is needed

**Review by gomathyk — APPROVED** [review]

**Pull request merged**